### PR TITLE
Adding route extractor

### PR DESCRIPTION
**Pull request by gwendolyngoetz**
### Description

Adding a class that can be used within custom Play Actions to pull out path parameter values. Play doesn't current provide an easy way to access these value natively.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Created unit and/or browser tests which fail without the change (if possible)

### Issue(s) this completes

Related to: #5541

### CODE REVIEW

File: server/app/actions/RouteExtractor.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ * <p>This is modified from the original. Created by Akinniranye James Ayodele on 8/12/16 12:56 PM.
+ * Project Andromeda Version 0.1
+ */
+public final class RouteExtractor {
+  private final String routePattern;
+  private final String path;
+  private final ImmutableMap<String, String> routeParameters;
+
+  // Example: something like /$id<[^/]+>
```
Comment (melanie-exygy) [inline server/app/actions/RouteExtractor.java:28]: suggestion (non-blocking):  I may not be the only one that doesn't know what this regex does off the top of my head.  Would you mind giving a one-line summary of what it does?

File: server/app/actions/RouteExtractor.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  public long getLongValue(String key) {
+    if (!routeParameters.containsKey(key)) {
+      throw new RuntimeException(String.format("Could not find '%s' in route '%s'", key, path));
+    }
+
+    try {
+      return Long.parseLong(routeParameters.get(key));
+    } catch (NumberFormatException ex) {
+      throw new RuntimeException(
+          String.format("Could parse value from '%s' in route '%s'", key, path), ex);
```
Comment (melanie-exygy) [inline server/app/actions/RouteExtractor.java:91]: nit:  I think the message should say "Could _not_ parse...")

File: server/app/actions/RouteExtractor.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+   */
+  public long getLongValue(String key) {
+    if (!routeParameters.containsKey(key)) {
+      throw new RuntimeException(String.format("Could not find '%s' in route '%s'", key, path));
+    }
+
+    try {
+      return Long.parseLong(routeParameters.get(key));
+    } catch (NumberFormatException ex) {
+      throw new RuntimeException(
```
Comment (melanie-exygy) [inline server/app/actions/RouteExtractor.java:90]: question (non-blocking):  Why not re-throw the more specific `NumberFormatException`?

Comment (gwendolyngoetz) [reply server/app/actions/RouteExtractor.java:91]: The `NumberFormatException` doesn't have a constructor to take an exception as a parameter. I don't want to lose the stack full stack trace.

**Review by melanie-exygy — APPROVED** [review]

**Pull request merged**